File: ts/src/contracts/tunnelServicePropertiesStatics.ts
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

import {
    TunnelServiceProperties as ITunnelServiceProperties,
    prodFirstPartyAppId,
    ppeFirstPartyAppId,
    devServiceAppId,
    prodThirdPartyAppId,
    ppeThirdPartyAppId,
    devThirdPartyAppId,
    prodGitHubAppClientId,
    ppeGitHubAppClientId,
    devGitHubAppClientId,
    localGitHubAppClientId,
    prodDnsName,
    ppeDnsName,
    devDnsName,
    localDnsName,
} from './tunnelServiceProperties';

/**
 * Gets production service properties.
 */
export const production = <ITunnelServiceProperties>{
    serviceUri: `https://${prodDnsName}/`,
    serviceAppId: prodFirstPartyAppId,
    serviceInternalAppId: prodThirdPartyAppId,
    gitHubAppClientId: prodGitHubAppClientId,
};

/**
 * Gets properties for the service in the staging environment (PPE).
 */
export const staging = <ITunnelServiceProperties>{
    serviceUri: `https://${ppeDnsName}/`,
    serviceAppId: ppeFirstPartyAppId,
    serviceInternalAppId: ppeThirdPartyAppId,
    gitHubAppClientId: ppeGitHubAppClientId,
};

/**
 * Gets properties for the service in the development environment.
 */
export const development = <ITunnelServiceProperties>{
    serviceUri: `https://${devDnsName}/`,
    serviceAppId: devServiceAppId,
    serviceInternalAppId: devThirdPartyAppId,
    gitHubAppClientId: devGitHubAppClientId,
};

/**
 * Gets properties for the service when running locally.
 *
 * Uses the same service app IDs as the development environment, but a different
 * GitHub app with localhost callback URLs.
 */
export const local = <ITunnelServiceProperties>{
    serviceUri: `https://${localDnsName}/`,
    serviceAppId: devServiceAppId,
    serviceInternalAppId: devThirdPartyAppId,
    gitHubAppClientId: localGitHubAppClientId,
};

/**
 * Gets properties for the service in the specified environment.
 */
export function environment(environmentName: string): ITunnelServiceProperties {
    if (!environmentName) {
        throw new Error(`Invalid argument: ${environmentName}`);
    }

    switch (environmentName.toLowerCase()) {
        case 'prod':
        case 'production':
            return production;
        case 'ppe':
        case 'preprod':
            return staging;
        case 'dev':
        case 'development':
            return development;
        case 'local':
            return local;
        default:
            throw new Error(`Invalid service environment: ${environmentName}`);
    }
}

File: java/src/main/java/com/microsoft/tunnels/contracts/TunnelServicePropertiesStatics.java
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

package com.microsoft.tunnels.contracts;

import java.util.Locale;

import org.apache.maven.shared.utils.StringUtils;

class TunnelServicePropertiesStatics {
  /**
   * Gets production service properties.
   */
  static final TunnelServiceProperties production = new TunnelServiceProperties(
      "https://" + TunnelServiceProperties.prodDnsName + "/",
      TunnelServiceProperties.prodFirstPartyAppId,
      TunnelServiceProperties.prodThirdPartyAppId,
      TunnelServiceProperties.prodGitHubAppClientId);

  /**
   * Gets properties for the service in the staging environment (PPE).
   */
  static final TunnelServiceProperties staging = new TunnelServiceProperties(
      "https://" + TunnelServiceProperties.ppeDnsName + "/",
      TunnelServiceProperties.ppeFirstPartyAppId,
      TunnelServiceProperties.ppeThirdPartyAppId,
      TunnelServiceProperties.ppeGitHubAppClientId);

  /**
   * Gets properties for the service in the development environment.
   */
  static final TunnelServiceProperties development = new TunnelServiceProperties(
      "https://" + TunnelServiceProperties.devDnsName + "/",
      TunnelServiceProperties.devServiceAppId,
      TunnelServiceProperties.devThirdPartyAppId,
      TunnelServiceProperties.devGitHubAppClientId);

  /**
   * Gets properties for the service when running locally.
   *
   * Uses the same service app IDs as the development environment, but a different
   * GitHub app with localhost callback URLs.
   */
  static final TunnelServiceProperties local = new TunnelServiceProperties(
      "https://" + TunnelServiceProperties.localDnsName + "/",
      TunnelServiceProperties.devServiceAppId,
      TunnelServiceProperties.devThirdPartyAppId,
      TunnelServiceProperties.localGitHubAppClientId);

  public static TunnelServiceProperties environment(String environmentName) {
    if (StringUtils.isBlank(environmentName)) {
      throw new IllegalArgumentException(environmentName);
    }

    switch (environmentName.toLowerCase(Locale.ROOT)) {
      case "prod":
      case "production":
        return TunnelServiceProperties.production;
      case "ppe":
      case "preprod":
      case "staging":
        return TunnelServiceProperties.staging;
      case "dev":
      case "development":
        return TunnelServiceProperties.development;
      case "local":
        return TunnelServiceProperties.local;
      default:
        throw new IllegalArgumentException("Invalid service environment: " + environmentName);
    }
  }
}
